### PR TITLE
Filelist mvvm

### LINKED ISSUE #467: Move FileListViewController to MVVM
- [ ] Explore realm "live objects" for file list ?
- [ ] Handle cases for other FileList types
- [ ] Also do #511

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**
Close #467

### CODE REVIEW

File: kDrive/UI/Controller/Files/File List/FileListViewController.swift [hunk 1]
```diff
@@ -0,0 +1,913 @@
...
+        viewModel.listStyleButtonPressed()
+    }
+
+    #if !ISEXTENSION
+        func uploadCardSelected() {
+            let uploadViewController = UploadQueueViewController.instantiate()
+            uploadViewController.currentDirectory = viewModel.currentDirectory
+            navigationController?.pushViewController(uploadViewController, animated: true)
+        }
+
```
Comment (flowbe) [inline kDrive/UI/Controller/Files/File List/FileListViewController.swift:762]: Small formatting issue

File: kDrive/UI/Controller/Photo/PhotoPickerDelegate.swift [hunk 2]
```diff
@@ -85,24 +87,23 @@ extension PhotoPickerDelegate: PHPickerViewControllerDelegate {
...
+                    guard !importedFiles.isEmpty else {
+                        return
                     }
-                } catch {
-                    DispatchQueue.main.async {
-                        self.handleError(error)
+                    do {
+                        try FileImportHelper.instance.upload(files: importedFiles, in: self.currentDirectory, drive: self.driveFileManager.drive)
+                        await self.showUploadSnackbar(count: importedFiles.count, filename: importedFiles[0].name)
+
```
Comment (flowbe) [inline kDrive/UI/Controller/Photo/PhotoPickerDelegate.swift:104]: Empty line

File: kDriveCore/Data/Cache/DriveFileManager.swift [hunk 3]
```diff
@@ -348,7 +348,7 @@ public class DriveFileManager {
             // Get children from API
             let children: [File]
             let responseAt: Int?
-            if directory.isRoot {
+            if directory.isRoot && !drive.sharedWithMe {
```
Comment (flowbe) [inline kDriveCore/Data/Cache/DriveFileManager.swift:351]: This can be removed

File: kDriveCore/Utils/Logging.swift [hunk 4]
```diff
@@ -61,7 +65,7 @@ public enum Logging {
     private static func initNetworkLogging() {
         #if DEBUG
             if !Constants.isInExtension {
-                Atlantis.start(hostName: ProcessInfo.processInfo.environment["hostname"])
+                Atlantis.start(hostName: "127.0.0.1")
```
Comment (flowbe) [inline kDriveCore/Utils/Logging.swift:68]: 😬

**Review by flowbe — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Review by Ambrdctr — APPROVED** [review]

**Review by flowbe — APPROVED** [review]

**Pull request merged**